lib/delayed/job_groups/job_group.rb

```diff
@@ -22,6 +22,9 @@ class JobGroup < ActiveRecord::Base
       has_many :queued_jobs, -> { where(failed_at: nil, locked_by: nil) }, class_name: '::Delayed::Job',
                 dependent: :delete_all
 
+      scope :ready, -> { where(queueing_complete: true, blocked: false) }
+      scope :with_no_open_jobs, -> { left_joins(:active_jobs).group(:id).having('count(delayed_jobs.id) == 0') }
```

Comment:
Have you checked to make sure this query runs reasonably fast where there are lots of jobs/job groups? It might be faster to use a `NOT EXISTS` query:

```sql
SELECT *
FROM delayed_job_groups
WHERE NOT EXISTS (SELECT * FROM delayed_jobs where delayed_jobs.job_group_id = delayed_job_groups.id)
```

Reply:
Heh, ok you're correct, the `NOT EXISTS` query is ~8-10x faster than my `having` query. I setup a benchmark in  to test this. It used 10,000 job groups each with 0-90 jobs.

```
       user     system      total        real
having 79.297660   0.334331  79.631991 ( 80.055589)
exists  8.398553   0.055814   8.454367 (  8.543264)
```

Updated to use `NOT EXISTS`.